pages/searching/dfs.tsx

```diff
@@ -1,10 +1,129 @@
       <Heading title="Depth-first Search" />
+      <Heading subHeading title="Facts">
+        <ul>
+          <li>A specific implementation of the General Graph Traversal Algorithm</li>
+          <li>Grey vertex selection rule: The next grey vertex is the youngest remaining grey vertex</li>
+          <li>Utilises a <Link href="/datastructures/stack"><a style={{ color: "#0000EE", textDecoration: "underline" }}>stack</a></Link> implementation to decide next node to visit</li>
+        </ul>
+      </Heading>
+      <Heading title="Pseudocode" />
+        <Heading subHeading title="DFS Pseudocode">
+          <Code text={dfsPseudocode} />
+        </Heading>
+        <Heading subHeading title="DFSVisit Iterative Pseudocode">
+          <Code text={dfsVisitIterativePseudocode} />
+        </Heading>
+        <Heading subHeading title="DFSVisit Recursive Pseudocode">
+          <Code text={dfsVisitRecursivePseudocode} />
+        </Heading>
+      <Heading title="Complexity" />
+      <StyledTable>
+          <TableHead>
+            <TableRow>
+              <StyledCell>Graph Data Structure</StyledCell>
+              <StyledCell>Running Time</StyledCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            <TableRow>
+              <StyledCell>Adjacency List</StyledCell>
+              <StyledCell>Θ(n + e) where e is the number of edges</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Adjacency Matrix</StyledCell>
+              <StyledCell>Θ(n) per vertex for n vertices = Θ(n²)</StyledCell>
+            </TableRow>
+          </TableBody>
+        </StyledTable>
+      <Heading title="Types of Arcs" />
+      Let (v, w) be an arc in a digraph traversed by DFS.
+      <StyledTable>
+          <TableHead>
+            <TableRow>
+              <StyledCell>Case</StyledCell>
+              <StyledCell>Condition</StyledCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            <TableRow>
+              <StyledCell>Tree Arc</StyledCell>
+              <StyledCell>seen[v] {"<"} seen[w] {"<"} done[w] {"<"} done[v]</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Forward Arc</StyledCell>
+              <StyledCell>seen[v] {"<"} seen[w] {"<"} done[w] {"<"} done[v]</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Back Arc</StyledCell>
+              <StyledCell>seen[w] {"<"} seen[v] {"<"} done[v] {"<"} done[w]</StyledCell>
+            </TableRow>
+            <TableRow>
+              <StyledCell>Cross Arc</StyledCell>
+              <StyledCell>seen[w] {"<"} done[w] {"<"} seen[v] {"<"} done[v]</StyledCell>
+            </TableRow>
+          </TableBody>
+        </StyledTable>
+        <Heading title="Determining Ancestors Using DFS" />
+        Let v and w be nodes in a search forest F obtained by performing DFS. Additionally suppose that seen[v] {"<"} seen[w]
```

Comment:
Could you wrap this in a `Typography` too.

Reply:
All done! :)